concordium-consensus/src/Concordium/GlobalState/PurgeTransactions.hs

```diff
@@ -188,3 +190,112 @@ purgeTables lastFinCommitPoint oldestArrivalTime currentTime TransactionTable{..
               _ttNonFinalizedTransactions = newNFT,
               _ttNonFinalizedChainUpdates = newNFCU
             }
+
+-- |Update the transaction table and pending transaction table as a result of constructing a block.
+-- The transactions added to the block are marked as committed.
+-- Failed transactions are purged from the transaction table if they are not committed since the
+-- last finalized block. The pending transaction table is updated to reflect the executed
+-- transactions and the purged transactions.
+--
+-- PRECONDITION: All of the filtered transactions are present in the transaction table and pending
+-- transaction table.
+filterTables ::
+    (IsCommitPoint cp) =>
+    -- |'CommitPoint' of the last finalized block
+    cp ->
+    -- |'CommitPoint' of block that transactions were added in
+    cp ->
+    -- |'BlockHash' of block that transactions were added in
+    BlockHash ->
+    -- |Filtered transactions as a result of constructing the block.
+    FilteredTransactions ->
+    -- |Transaction table to update
+    TransactionTable ->
+    -- |Pending transaction table to update
+    PendingTransactionTable ->
+    (TransactionTable, PendingTransactionTable)
+filterTables lastFinCommit commitAt commitBlock FilteredTransactions{..} tt0 ptt0 =
+    purgeUpdates . purgeCredentials . purgeTransactions $
+        (commitAdded tt0, forwarded)
+  where
+    addedTransactions = fst . fst <$> ftAdded
+    -- Mark added transactions as committed
+    commitAdded = commitAdded' 0 addedTransactions
+    commitAdded' _ [] tt = tt
+    commitAdded' i (bi : bis) tt =
+        commitAdded' (i + 1) bis $!
+            tt & ttHashMap . ix (getHash bi) . _2 %~ addResult commitBlock commitAt i
+    -- Update the pending transaction table
+    forwarded = forwardPTT addedTransactions ptt0
+    -- Determine if a block item is committed to a potentially-live block.
+    isCommitted :: TransactionTable -> WithMetadata a -> Bool
+    isCommitted tt t = case tt ^? ttHashMap . ix (wmdHash t) . _2 . tsCommitPoint of
+        Just transactionCP -> commitPoint lastFinCommit < commitPoint transactionCP
```

Comment:
It's with the perspective on whether it's comitted to a **live** block or not. So the commit point must be strictly larger than the last finalized commit point?

But yes you are right, if it was `<=` it would still hit the `Nothing` case as you are correct regarding whether the finalized transactions are here or not, so the result of the function will be the same.